dotcom-rendering/src/web/lib/useSDC.tsx

```diff
@@ -0,0 +1,40 @@
+import type { ModuleDataResponse } from '@guardian/support-dotcom-components';
+import {
+	getBanner,
+	getEpic,
+	getLiveblogEpic,
+	getPuzzlesBanner,
+} from '@guardian/support-dotcom-components';
+import type {
+	BannerPayload,
+	EpicPayload,
+} from '@guardian/support-dotcom-components/dist/dotcom/src/types';
+import type { SWRConfiguration } from 'swr';
+import useSWR from 'swr';
+
+const options: SWRConfiguration = {
+	revalidateIfStale: false,
+	revalidateOnFocus: false,
+	revalidateOnReconnect: false,
+};
+
+/**
+ * Hooks for making requests to the support-dotcom-components API
+ */
+type UseSDC<T> = (
+	baseUrl: string,
+	payload: T,
+) => ModuleDataResponse | undefined;
+
+export const useSDCEpic: UseSDC<EpicPayload> = (baseUrl, payload) =>
+	useSWR('epic', () => getEpic(baseUrl, payload), options).data;
```

Comment:
`useSWR` also returns an `.error` property. I wonder if the underlying request fails would we want to log that somehow? Or perhaps that's already handled upstream (in `useEpic` for example)?

Reply:
the functions from SDC just throw an error - https://github.com/guardian/support-dotcom-components/blob/main/packages/dotcom/src/requests.ts#L49
I'll have a go at logging errors here